test/e2e/nvidia-container-toolkit_test.go

```diff
@@ -0,0 +1,106 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package e2e
+
+import (
+	"context"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+// Actual test suite
+var _ = Describe("NVIDIA Container Toolkit", func() {
+	JustBeforeEach(func(ctx context.Context) {
+		// Placeholder for any setup that needs to be done before each test
+	})
+
+	AfterEach(func(ctx context.Context) {
+		// Placeholder for any cleanup that needs to be done after each test
+	})
+
+	// GPUs are accessible in a container: Running nvidia-smi -L inside the
+	// container shows the same output inside the container as outside the
+	// container. This means that the following commands must all produce
+	// the same output
+	When("Running nvidia-smi -L", Ordered, func() {
+		It("it should show the same output from outside a container", func(ctx context.Context) {
+			// collect the output of nvidia-smi -L
+			rawOut, err := runScript("nvidia-smi -L")
+			Expect(err).ToNot(HaveOccurred())
+
+			// collect the output of nvidia-smi -L inside a container with --gpus all and NVIDIA_VISIBLE_DEVICES=all
+			outContainer, err := runScript("docker run --rm -i --runtime=nvidia -e NVIDIA_VISIBLE_DEVICES=all ubuntu nvidia-smi -L")
+			Expect(err).ToNot(HaveOccurred())
+
+			// compare the outputs
+			Expect(outContainer).To(Equal(rawOut))
+
+			// collect the output of nvidia-smi -L inside a container with --gpus all and NVIDIA_VISIBLE_DEVICES=runtime.nvidia.com/gpu=all
+			outContainer, err = runScript("docker run --rm -i --runtime=nvidia -e NVIDIA_VISIBLE_DEVICES=runtime.nvidia.com/gpu=all ubuntu nvidia-smi -L")
+			Expect(err).ToNot(HaveOccurred())
+
+			// compare the outputs
+			Expect(outContainer).To(Equal(rawOut))
+
+			// collect the output of nvidia-smi -L inside a container with --gpus all
+			outContainer, err = runScript("docker run --rm -i --runtime=nvidia --gpus all ubuntu nvidia-smi -L")
+			Expect(err).ToNot(HaveOccurred())
+
+			// compare the outputs
+			Expect(outContainer).To(Equal(rawOut))
+		})
+	})
+
+	// A vectorAdd sample runs in a container with access to all GPUs.
+	// The following should all produce the same result.
+	When("Running the cuda-vectorAdd sample", Ordered, func() {
+		It("it should show the same output from each run", func(ctx context.Context) {
+			out1, err := runScript("docker run --rm -i --runtime=nvidia -e NVIDIA_VISIBLE_DEVICES=all nvcr.io/nvidia/k8s/cuda-sample:vectoradd-cuda12.5.0")
+			Expect(err).ToNot(HaveOccurred())
```

Comment:
Question: in the failures that we were seeing when MIG was enabled on a device, was that reported as an error here?

Reply:
didn't saw it, I was manually running the command first before trying it from `go test`.